OrbitLinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -145,7 +147,9 @@ UprobesUnwindingVisitor::CallstackFramesFromInstructionPointers(
     return std::vector<CallstackFrame>();
   }
   std::vector<CallstackFrame> callstack_frames;
-  for (uint64_t i = 0; i < size; i++) {
+  // The top of a callchain is always inside the kernel code.
```

Comment:
Is this the right function for this? This function performs the transformation to a `std::vector<CallstackFrame>`. It seems that other kinds of processing should be done by the caller or somewhere else.

Reply:
Maybe just 
```
return_address_manager_.PatchCallchain(event->GetTid(), event->GetCallchain() + 1,
                                         event->GetCallchainSize() - 1,
                                         current_maps_.get());
```
with a comment explaining it (assuming it's already checked somewhere that the callchain is not empty).

Reply:
Thanks. I moved the stripping of the callchain to the caller, as you suggested.